solidity/test/maintainer/MaintainerProxy.test.ts

```diff
@@ -0,0 +1,3434 @@
+        await restoreSnapshot()
+      })
+
+      context("when the main UTXO parameter is valid", () => {
+        context("when the balance is below the dust threshold", () => {
+          const mainUtxo = {
+            txHash: ethers.constants.HashZero,
+            txOutputIndex: 0,
+            txOutputValue: constants.movingFundsDustThreshold - 1,
+          }
+
+          let tx: ContractTransaction
+          let initThirdPartyBalance: BigNumber
+
+          before(async () => {
+            await createSnapshot()
+
+            await bridge.setWalletMainUtxo(
+              ecdsaWalletTestData.pubKeyHash160,
+              mainUtxo
+            )
+
+            initThirdPartyBalance = await provider.getBalance(
+              authorizedMaintainer.address
+            )
+
+            tx = await maintainerProxy
+              .connect(authorizedMaintainer)
+              .notifyMovingFundsBelowDust(
+                ecdsaWalletTestData.pubKeyHash160,
+                mainUtxo
+              )
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should not revert", async () => {
+            await expect(tx.wait()).not.to.be.reverted
+          })
+
+          it("should refund ETH", async () => {
+            const postThirdPartyBalance = await provider.getBalance(
+              authorizedMaintainer.address
+            )
+            const diff = postThirdPartyBalance.sub(initThirdPartyBalance)
+
+            expect(diff).to.be.gt(0)
+            expect(diff).to.be.lt(
+              ethers.utils.parseUnits("1000000", "gwei") // 0,001 ETH
+            )
+          })
+        })
+      })
+    })
+  })
+
+  describe("submitMovedFundsSweepProof", () => {
+    context("when called by an unauthorized third party", async () => {
+      it("should revert", async () => {
+        const data: MovedFundsSweepTestData = MovedFundsSweepWithoutMainUtxo
+        const tx = maintainerProxy
+          .connect(thirdParty)
+          .submitMovedFundsSweepProof(
+            data.sweepTx,
+            data.sweepProof,
+            data.mainUtxo
+          )
+        await expect(tx).to.be.revertedWith("Caller is not authorized")
+      })
+    })
+
+    context("when transaction proof is valid", () => {
+      before(async () => {
+        await createSnapshot()
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+      context("when there is only one output", () => {
+        context("when sweeping wallet is in the Live state", () => {
+          context("when main UTXO data are valid", () => {
+            context(
+              "when transaction fee does not exceed the sweep transaction maximum fee",
+              () => {
+                context("when the sweeping wallet has no main UTXO set", () => {
+                  context(
+                    "when there is a single input referring to a Pending sweep request",
+                    () => {
```

Comment:
Can we reduce it to just:
```
context("when there is only one output", () => {
  context("when the sweeping wallet has no main UTXO set", () => {
  }

  context("when the sweeping wallet has a main UTXO set", () => {
  }
}
```

This is the only information that is relevant in terms of gas refunds.